File: DP/javascript/2225.js
const [n, k] = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .split(" ")
  .map(Number);
const dp = Array.from(Array(201), () => new Array(201).fill(1));

for (let i = 1; i <= 200; i++) {
  for (let j = 2; j <= 200; j++) {
    dp[i][j] = (dp[i][j - 1] + dp[i - 1][j]) % 1000000000;
  }
}
console.log(dp[n][k]);

File: DP/python/2225.py
n,k = map(int,input().split())
dp = [[1 for _ in range(201)] for _ in range(201)]
for i in range(1,201):
  for j in range(2,201):
    dp[i][j] = (dp[i][j-1] + dp[i-1][j])%1000000000
print(dp[n][k])